extensions/extensions-configuration-store-vault/src/main/java/com/boozallen/aissemble/configuration/dao/PropertyDao.java

```diff
@@ -0,0 +1,150 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class PropertyDao implements Dao<Property> {
+    private static final Logger logger = LoggerFactory.getLogger(PropertyDao.class);
+
+    @Override
+    public boolean checkEmpty() {
+        throw new RuntimeException("not implemented");
+    }
+
+    /**
+     * Read the property from Vault server with given group name and property name
+     * @param groupName to read from the vault
+     * @param propertyName to read from the vault
+     */
+    @Override
+    public Property read(String groupName, String propertyName) {
+        try {
+            final Vault vault = getVault();
+            // read from Vault
+            final LogicalResponse readResponse = vault.logical()
+                    .read(String.format("properties/%s/%s", groupName, propertyName));
+
+            return new Property(groupName, propertyName, readResponse.getData().get(propertyName));
+        } catch (VaultException e) {
+            throw new PropertyDaoException(String.format("Unable to read from vault with groupName: %s and property name: %s.", groupName, propertyName), e);
+        }
+    }
+
+    /**
+     * Write given property to the vault
+     * @param property to be written to vault
+     * @throws PropertyDaoException
+     */
+    @Override
+    public void write(Property property) throws PropertyDaoException {
+        if (property != null) {
+            try {
+                final Vault vault = getVault();
+                this.write(vault, Collections.singletonList(property));
+            } catch (VaultException e) {
+                throw new PropertyDaoException(String.format("Unable to write to vault with property: %s.", property), e);
+            }
+        } else {
+            throw new PropertyDaoException("Unable to write to vault with property: null");
+        }
+    }
+
+    /**
+     * Write given set of properties to the vault
+     * @param properties to be written to vault
+     * @throws PropertyDaoException
+     */
+    @Override
+    public void write(Set<Property> properties) throws PropertyDaoException {
+        if (properties != null && !properties.isEmpty()) {
+            try {
+                final Vault vault = getVault();
+                this.write(vault, new ArrayList<>(properties));
+            } catch (VaultException e) {
+                throw new PropertyDaoException(String.format("Unable to write to vault with properties: %s.", properties), e);
+            }
+        } else {
+            throw new PropertyDaoException("Unable to write to vault with empty properties");
+        }
+
+    }
+
+    private void write(Vault vault, List<Property> properties) throws VaultException {
+        for (Property property : properties) {
+            // path: properties/<<group name>>
+            String path = String.format("properties/%s", property.getGroupName());
```

Comment:
A: I think we should use a less generic name for the top-level.  The vault instance may be used for more than just aissemble-based functionality. Maybe `aissemble-properties`.